test/functional/ui/screen.lua

```diff
@@ -181,33 +193,53 @@ function Screen:set_hlstate_cterm(val)
   self._hlstate_cterm = val
 end
 
-function Screen:attach(options)
+function Screen:attach(options, session)
+  if session == nil then
+    session = get_session()
+  end
   if options == nil then
     options = {}
   end
   if options.ext_newgrid == nil then
     options.ext_newgrid = true
   end
+  if options.ext_multigrid then
+    options.ext_newgrid = true
+    self._multigrid = true
+  end
+
+  self._session = session
   self._options = options
   self._clear_attrs = (options.ext_newgrid and {{},{}}) or {}
-  uimeths.attach(self._width, self._height, options)
+  self.uimeths.attach(self._width, self._height, options)
   if self._options.rgb == nil then
     -- nvim defaults to rgb=true internally,
     -- simplify test code by doing the same.
     self._options.rgb = true
   end
+  self._session = session
 end
 
 function Screen:detach()
-  uimeths.detach()
+  self.uimeths.detach()
+  self._session = nil
 end
 
 function Screen:try_resize(columns, rows)
-  uimeths.try_resize(columns, rows)
+  self.uimeths.try_resize(columns, rows)
+  -- Give ourselves a chance to _handle_resize, which requires using
+  -- self.sleep() (for the resize notification) rather than run()
+  self:sleep(0.1)
+end
+
+function Screen:try_resize_grid(grid, columns, rows)
+  self._grid = self._grids[1]
```

Comment:
This is not needed here. Only legacy grid code uses it, and for it it is set in (legacy) `_handle_resize`.